data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -169,430 +143,221 @@ def get_table_dates_info(project_id, dataset_id, client):
         formatter_class=argparse.RawDescriptionHelpFormatter)
     parser.add_argument('-p',
-                        '--project-id',
+                        '--project_id',
                         action='store',
                         dest='project_id',
                         help='Identifies the project to retract data from',
                         required=True)
-    parser.add_argument('-t',
-                        '--ticket-number',
+    parser.add_argument('-d',
+                        '--dataset_ids',
                         action='store',
-                        dest='ticket_number',
-                        help='Ticket number to append to sandbox table names',
+                        nargs='+',
+                        dest='dataset_ids',
+                        help='Identifies datasets to target. Set to '
+                        '"all_datasets" to target all datasets in project '
+                        'or specific datasets as -d dataset_1 dataset_2 etc.',
                         required=True)
-    parser.add_argument(
-        '-i',
-        '--pids-project-id',
-        action='store',
-        dest='pids_project_id',
-        help='Identifies the project where the pids table is stored',
-        required=True)
-    parser.add_argument(
-        '-d',
-        '--pids-dataset-id',
-        action='store',
-        dest='pids_dataset_id',
-        help='Identifies the dataset where the pids table is stored',
-        required=True)
-    parser.add_argument('-s',
-                        '--pids-table',
+    parser.add_argument('-a',
+                        '--fq_deact_table',
                         action='store',
-                        dest='pids_table',
-                        help='Identifies the table where the pids are stored',
+                        dest='fq_deact_table',
+                        help='Specify fully qualified deactivated table '
+                        'as "project.dataset.table"',
                         required=True)
-    parser.add_argument(
-        '-l',
-        '--dataset_list',
-        dest='dataset_list',
-        action='append',
-        required=False,
-        help=
-        'Optional parameter, list of datasets to run retraction on vs. entire project send '
-        'multiple as separate argument ie: -l dataset_1 -l dataset_2')
-    parser.add_argument('-c',
-                        '--console-log',
-                        dest='console_log',
-                        action='store_true',
-                        required=False,
-                        help='Log to the console as well as to a file.')
-    return parser.parse_args(raw_args)
-
-
-def main(args=None):
+    parser.add_argument('-r',
+                        '--fq_pid_rid_table',
+                        action='store',
+                        dest='fq_pid_rid_table',
+                        help='Specify fully qualified pid-rid mapping table '
+                        'as "project.dataset.table"',
```

Comment:
Based on this help text, consider adding a type check argument to this parameter.  That way this can fail fast if the parameter doesn't parse as expected.  Python describes adding user defined functions for this [here](https://docs.python.org/3/library/argparse.html#type).  It's been done in the curation project before, [here](https://github.com/all-of-us/curation/blob/develop/data_steward/tools/run_deid.py#L193).

Reply:
Thanks for the links, I've added this check.